caps/types_test.go

```diff
@@ -50,17 +53,34 @@ func (s *TypeSuite) TestValidateOK(c *C) {
 	c.Assert(err, IsNil)
 }
 
-func (s *TypeSuite) TestValidateAttributes(c *C) {
+func (s *TypeSuite) TestValidateAttributesUnexpectedAttrs(c *C) {
+	t := &Type{
+		Name:          "t",
+		RequiredAttrs: nil,
+	}
 	cap := &Capability{
 		Name:  "name",
 		Label: "label",
-		Type:  testType,
-		Attrs: map[string]string{
-			"Key": "Value",
-		},
+		Type:  t,
+		Attrs: map[string]string{"k": "v"},
 	}
-	err := testType.Validate(cap)
-	c.Assert(err, ErrorMatches, "attributes must be empty for now")
+	err := t.Validate(cap)
+	c.Assert(err, ErrorMatches, `capability contains unexpected attribute "k"`)
+}
+
+func (s *TypeSuite) TestValidateAttributesRequiredAttrs(c *C) {
+	t := &Type{
+		Name:          "t",
+		RequiredAttrs: []string{"k"},
+	}
+	cap := &Capability{
+		Name:  "name",
+		Label: "label",
+		Type:  t,
+		Attrs: map[string]string{},
```

Comment:
No need, removed now.

The second part is fixed with the exception that the last test you've mentioned is not implemented (since I didn't change Validate() yet as I need to wait for your explanation).